app/services/whatsapp_service.py

```diff
@@ -38,6 +44,123 @@ async def send_message(
+
+            payload = generate_payload_for_image(wa_id, media_id, caption)
+
+            response = await self.client.post(
+                "/messages", json=payload, headers=self.headers
+            )
+            log_httpx_response(response)
+            response.raise_for_status()
+            message_sent_successfully = True
+            
+        except httpx.RequestError as e:
+            self.logger.error("Image Message Request Error: %s", e)
+            
+        except Exception as e:
+            self.logger.error("Image Message Unexpected Error: %s", e)
+            
+        finally:
+            if message_sent_successfully:
+                await self.delete_media(media_id,image_path)  # Clean up media after sending
+
+
+    async def delete_media(self, media_id: str, image_path: str) -> None:
+        """Delete upload media from WhatsApp and locally"""         
+        if settings.mock_whatsapp:
+            self.logger.info("Mock delete media called for media id %s", media_id)
+            return
+
+        try:
+            headers = {
+                "Authorization": self.headers["Authorization"],
+            }
+            url = f"https://graph.facebook.com/{settings.meta_api_version}/{media_id}"
+            response = await self.client.delete(url, headers=headers)
+            log_httpx_response(response)
+            response.raise_for_status()
+        except httpx.RequestError as e:
+            self.logger.error("Media Delete Request Error: %s", e)
+            raise
+        except Exception as e:
+            self.logger.error("Media Delete Unexpected Error: %s", e)
+            raise
+        
+        try:
+            os.remove(image_path)
+        except Exception as e:
+            self.logger.error(f"Failed to delete file {image_path}: {e}")
+
+    async def upload_media(self, path: str, img_type: str) -> Optional[str]:
+        """Upload an image to WhatsApp and return the media ID."""
+
+        if settings.mock_whatsapp:
+            self.logger.info("Mock upload media called for path %s", path)
+            return None
+
+        file_path = Path(path)
+
+        if not file_path.is_file():
+            raise FileNotFoundError(f"Image file not found at {path}")
+
+        file_size = file_path.stat().st_size
+        if file_size > self._MAX_IMAGE_SIZE_BYTES:
+            raise ValueError(
+                "Image size exceeds 5 MB limit for WhatsApp media uploads."
+            )
+
+        try:
+            with file_path.open("rb") as file_handle:
+                files = {
+                    "file": (file_path.name, file_handle, img_type),
+                }
+                data = {"messaging_product": "whatsapp"}
+                headers = {
+                    "Authorization": self.headers["Authorization"],
+                }
+                response = await self.client.post(
+                    "/media", data=data, files=files, headers=headers
+                )
```

Comment:
Do we have to make the post inside the with? Couldn't we just do:

```suggestion
            data = {"messaging_product": "whatsapp"}
            headers = {"Authorization": self.headers["Authorization"]}
            with file_path.open("rb") as file_handle:
                files = {"file": (file_path.name, file_handle, img_type)}

            response = await self.client.post("/media", data=data, files=files, headers=headers)
```

Reply:
Isn't this unecessary memoryalocation? we save a copy of the image in the memory. If the image is 800 k pixels (hehehe) it might not be very good